pipeline/stage03_trigger_detection/scripts/minima.py

```diff
@@ -6,17 +6,61 @@
 from utils import load_neo, write_neo, remove_annotations
 
 
-def detect_minima(asig, order):
+def detect_minima(asig, order, interpolation_points, interpolation):
     signal = asig.as_array()
+    sampling_time = asig.times[1] - asig.times[0]
+
     t_idx, channel_idx = argrelmin(signal, order=order, axis=0)
+    
+    if interpolation:
+
+        fitted_idx_times = np.zeros([len(t_idx)])
+        
+        start_arr = t_idx - int(interpolation_points/2)
+        start_arr = np.where(start_arr > 0, start_arr, 0)
+        stop_arr = start_arr + int(interpolation_points)
+        stop_arr = np.where(stop_arr < len(signal), stop_arr, len(signal))
```

Comment:
the alternative value needs to be `len(signal)-1`, since arrays are indexed from 0, len(array) is already out of bounds